cli/command/registry/warning.go

```diff
@@ -0,0 +1,22 @@
+package registry
+
+import (
+	"os"
+
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/docker/cli/cli/streams"
+	"github.com/docker/cli/internal/tui"
+)
+
+type outErr interface {
+	Err() *streams.Out
+}
+
+// maybePrintEnvAuthWarning if the `DOCKER_AUTH_CONFIG` environment variable is
+// set this function will output a warning to stdErr
+func maybePrintEnvAuthWarning(out outErr) {
```

Comment:
If you change this to accept `command.Streams`, we don't need the custom interface;


```suggestion
func maybePrintEnvAuthWarning(out command.Streams) {
```